flocker/control/test/test_model_change.py

```diff
@@ -0,0 +1,418 @@
+
+        class Different(CheckedPMap):
+            __key_type__ = ChangedSubtype
+            __value_type__ = int
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+    def test_pmap_value_type_changed(self):
+        """
+        If the type of the value of a ``PMap`` is the same, but it has
+        internally changed then the output changes.
+        """
+        class Original(CheckedPMap):
+            __key_type__ = int
+            __value_type__ = Subtype
+
+        class Different(CheckedPMap):
+            __key_type__ = int
+            __value_type__ = ChangedSubtype
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+    def test_pset_value_new_type(self):
+        """
+        If the type of the value of a ``PSet`` changes the output changes.
+        """
+        class Original(CheckedPSet):
+            __type__ = int
+
+        class Different(CheckedPSet):
+            __type__ = str
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+    def test_pset_value_type_changed(self):
+        """
+        If the type of the value of a ``PSet`` is the same, but it has
+        internally changed then the output changes.
+        """
+        class Original(CheckedPSet):
+            __type__ = Subtype
+
+        class Different(CheckedPSet):
+            __type__ = ChangedSubtype
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+    def test_pvector_value_new_type(self):
+        """
+        If the type of the value of a ``PVector`` changes the output changes.
+        """
+        class Original(CheckedPVector):
+            __type__ = int
+
+        class Different(CheckedPVector):
+            __type__ = str
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+    def test_pvector_value_type_changed(self):
+        """
+        If the type of the value of a ``PVector`` is the same, but it has
+        internally changed then the output changes.
+        """
+        class Original(CheckedPVector):
+            __type__ = Subtype
+
+        class Different(CheckedPVector):
+            __type__ = ChangedSubtype
+        Different.__name__ = "Original"
+
+        self.assert_catches_changes(Original, Different)
+
+
+def persist_model():
+    """
+    Store the current model to disk.
+
+    We also store the git hash of current checkout, so it's clear what
+    version of code was used to generate the model.
+    """
+    git_hash = check_output(
+        [b"git", b"show", b'--format=%H', b"-s"]).strip()
+    model = generate_model()
+    PERSISTED_MODEL.setContent(dumps(
+        {u"git_hash": git_hash, u"model": model}, sort_keys=True, indent=4,
+        separators=(',', ': ')))
+    print("Persisted model for git hash {}.".format(git_hash))
```

Comment:
:question: Meh `print`, mucking up stdout.  Maybe this could at least be in the `__main__` block so one can call this function without dumping out this user feedback, if one wanted.